src/scripts/main.js

```diff
@@ -1,3 +1,264 @@
+    errorMessages.push('Age range should be from 18 to 90.');
+  }
+
+  if (
+    employeeName === ''
+    || position === ''
+    || office === ''
+    || age === ''
+    || salary === ''
+  ) {
+    errorMessages.push('All fields are required.');
+  }
+
+  if (salary < 0) {
+    errorMessages.push('Salary should be larger than 0.');
+  }
+
+  if (Number.isNaN(salary)) {
+    errorMessages.push('Salary should be numeric value.');
+  }
+
+  return errorMessages;
+}
+
+const form = document.forms[0];
+
+form.addEventListener('submit', (handler) => {
+  handler.preventDefault();
+
+  const data = new FormData(form);
+  const salary = data.get('salary');
+  const employeeName = data.get('name');
+  const position = data.get('position');
+  const office = data.get('office');
+  const age = data.get('age');
+  const salaryString = `$` + Number(salary).toLocaleString();
+  const errorMessages = checkForErrors(
+    employeeName, position, office, age, salary
+  );
+
+  if (errorMessages.length) {
+    pushNotification(
+      10, 10, 'There was an error:', errorMessages.join('\n'), 'error'
+    );
+  } else {
+    pushNotification(
+      10, 10, 'Success', 'Employee was saved successfully.', 'success'
+    );
+
+    tableBody.insertAdjacentHTML('beforeend', `
+    <tr>
+      <td>${employeeName}</td>
+      <td>${position}</td>
+      <td>${office}</td>
+      <td>${age}</td>
+      <td>${salaryString}</td>
+    </tr>
+    `);
+
+    form.reset();
+  }
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const cell = e.target;
+  const cellIndex = cell.cellIndex;
+  const initialValue = cell.innerText;
+  const selectedRow = e.target.closest('tr');
+  const input = document.createElement('input');
+
+  input.classList.add('cell-input');
+  input.type = 'text';
+  input.name = 'cell-input';
+  input.setAttribute('value', initialValue);
+  cell.innerText = '';
+  cell.append(input);
+  input.focus();
+
+  input.addEventListener('keyup', (f) => {
+    if (f.key === 'Enter') {
+      updateCell();
+    }
+  });
+
+  input.addEventListener('blur', () => {
+    updateCell();
+  });
+
+  function updateCell() {
+    const cellValues = [];
+
+    for (let i = 0; i < 5; i++) {
+      if (cellIndex === i) {
+        cellValues.push(input.value);
+      } else {
+        cellValues.push(selectedRow.children[i].innerText);
+      }
+    }
+
+    if (cellValues[4][0] === '$') {
+      cellValues[4] = +cellValues[4].slice(1).replace(',', '');
+    } else {
+      cellValues[4] = +cellValues[4].replace(',', '');
+    }
+
+    selectedRow.classList.remove('active');
+
+    const errorMessages = checkForErrors(...cellValues);
+
+    if (errorMessages.length) {
+      cell.innerText = initialValue;
+
+      pushNotification(
+        10, 10, 'There was an error:', errorMessages.join('\n'), 'error'
+      );
+    } else {
+      if (cellIndex === 4) {
+        cell.innerText = `$` + Number(cellValues[4]).toLocaleString();
```

Comment:
it should be saved in the same format:
![image](https://user-images.githubusercontent.com/86923089/207421991-dcd8fa98-49d9-4933-a312-12ef5dd6a871.png)
there should be `,` instead of space ` `

Reply:
Then after editing and blurring this cell we'll get the error:
![image](https://user-images.githubusercontent.com/86923089/207422693-54197ac4-4845-4d2d-9a8c-c2efc509fd5c.png)

Also, it breaks down sorting

Reply:
I specified 'en-US' as locales parameter for the toLocaleString() method. I couldn't see this bug since my OS has English settings.

Reply:
Also added toLocaleString('en-US') here.